### ISSUE TITLE
Add Varnish support

### ISSUE THREAD

**Issue by Intrepidity** [fideloper/Vaprobash#440]
It would be nice to have out of the box support for Varnish, and having it listen on port 80 and move apache or nginx to 8080 or something.

**Comment by fideloper** [comment]
I've had mixed feeling since it's weird to have a proxy in place for development, but would be nice to have around so people can see an example of it in use. (Configuration for an HTTP cache is usually very specific to a user's application, however).

**Comment by Intrepidity** [comment]
I see your point, except there's also a big market for deploying vagrant boxes to production enviroments (digital ocean for example) or internally as intranet servers. So in my opinion there's definately a use-case to install Varnish. Configuration will indeed be difficult, but we can think of a sane default and maybe create a possibility for users to specify their custom vcs-file in a configuration option?

I can probably whip something up over the next 2 weeks.

**Comment by fideloper** [comment]
I agree, likely is a decent chunk of people wanting that. Vapro has never
been a setup for production (has debugging on, zero security in place), but
I do use it often to refer to scripts to perform tasks in production :D

Varnish is likely a good addition here.
